example/vm/config.yaml

```diff
@@ -134,14 +134,17 @@ processors:
         # For more details, see:
         # https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/attributesprocessor
         actions:
-            # In this case, the attributes processor maps the `service_version` attribute attached
-            # to telemetry from the prometheus/self receiver to `service.version` following OTel
-            # semantic conventions for service attributes.
+            # In this case, the attributes processor maps service attributes attached to telemetry
+            # from the prometheus/self receiver to service attributes following OTel semantic
+            # conventions for service attributes.
             # For more details, see:
             # https://opentelemetry.io/docs/specs/semconv/attributes-registry/service/
             - key: service.version
               action: insert
               from_value: service_version
+            - key: service.name
+              action: upsert
+              from_value: collector_name
```

Comment:
were you able to test this worked locally?

Reply:
Ah good catch, switch from_value to from_attribute. This is from my dev-adil project in public:

<img width="448" alt="image" src="https://github.com/lightstep/otel-collector-charts/assets/87725682/882d4496-b860-493c-979d-c3e24812b366">

Reply:
@jaronoff97 the reason I went with `collector_name` over `service_name` was to use the "good" name we chose instead of the prometheus scrape job name.

My sense was that this only will apply to the collector's meta metrics and not other metrics that flow through the collector because the label is added as part of the prom scrape config here: https://github.com/lightstep/otel-collector-charts/blob/60a781950ebaa81b921273a685a62dc66f92ff9b/example/vm/config.yaml#L7-L17

Phrasing this as a question: what is the value or reason in adding the `collector_name: sn-collector` label to the prometheus/self receiver's scrape config if we don't promote it into `service.name`? I'm genuinely asking out of curiosity, I don't really care all that much which attribute we copy into `service.name`

Reply:
ideally it would help you understand the source of the metric, a user may want to know "what actually scraped this metric" and you would use the collector_name for that. The reason i wanted us to use service_name here is that a user is probably going to use prometheus to scrape other things on a VM beyond the self scrape and this will help make sure all of their other metrics are tagged with service.name now.

Reply:
Got it, makes sense. Thanks!